drivers/crypto/crypto_stm32.c

```diff
@@ -75,17 +115,22 @@ static int do_encrypt(struct cipher_ctx *ctx, uint8_t *in_buf, int in_len,
 
 	k_sem_take(&data->device_sem, K_FOREVER);
 
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_cryp) || DT_HAS_COMPAT_STATUS_OKAY(st_stm32_aes)
 	status = HAL_CRYP_SetConfig(&data->hcryp, &session->config);
 	if (status != HAL_OK) {
 		LOG_ERR("Configuration error");
 		k_sem_give(&data->device_sem);
 		return -EIO;
 	}
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32l4_aes)
+	/* Device is initialized from the configuration in the encryption/decryption function
+	 * called bellow. */
+	memcpy(&data->hcryp.Init, &session->config, sizeof(session->config));
+#endif
 
-	status = HAL_CRYP_Encrypt(&data->hcryp, (uint32_t *)in_buf, in_len,
-				  (uint32_t *)out_buf, HAL_MAX_DELAY);
+	status = fn(&data->hcryp, in_buf, in_len, out_buf, HAL_MAX_DELAY);
 	if (status != HAL_OK) {
-		LOG_ERR("Encryption error");
+		LOG_ERR("Encryption/decryption error");
```

Comment:
Split overall changes added to the driver (which will impact already supported IPs) and L4 support additions in distinct commits.